### PR TITLE
Add maintenance notice to the README

### PR DESCRIPTION
**Pull request by binford2k**
r10k is now a mature product. This is intended to clarify our maintenance
committment and roadmap going forward. Approved by @david-piekny

### CODE REVIEW

File: README.mkd
Comment (mwaggett) [inline README.mkd]: I don't think this needs to be in a code block?

File: README.mkd
Comment (mwaggett) [inline README.mkd]: I don't even know if we should mention this part about new feature development - it would still require a Puppet engineer to review, merge, release the change, and make sure it doesn't negatively impact PE, and it sounds like that won't happen unless Product prioritizes it.

Comment (mwaggett) [reply README.mkd]: @binford2k ^this point still stands. I don't want to set the expectation that we will be merging community contributions on a regular basis.

Comment (themicahwilson) [reply README.mkd]: @mwaggett I think the message is fine as is. Let's get it up, can you approve it? Any open issues can be closed out with a "won't do", but it's not a priority to do this, it's really if someone has time and wants to. Looks like https://github.com/puppetlabs/r10k/pull/1315 and https://github.com/puppetlabs/r10k/pull/1292 might be worth pushing through, but it's not a priority, so if @binford2k has time, awesome, otherwise we will leave them.

Comment (mwaggett) [reply README.mkd]: The hesitation I have with this message is that "peer-reviewed community contributions, with the onus on the contributor and community for validation" still requires a Puppet engineer to review and merge the PR and confirm that it doesn't negatively impact PE or CD4PE. So any such PRs will continue to sit open unless you continue to monitor this repo and let us know when we should review those PRs. If that's part of the plan, I am happy to merge this, but if not, I don't want to set the set the expectation that we will merge PRs purely on the faith of the community, since the community can't validate against PE.

Comment (mwaggett) [reply README.mkd]: okay @themicahwilson wants this left as-is, so merging.

**Review by mwaggett — APPROVED** [review]

**Pull request merged**
